### ISSUE TITLE
Create Transactions Type MDM endpoint 

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#1786]
### Definition of Ready
- [x] Does this meet the INVEST criteria?
- [x] Does this have a clearly defined user?
- [x] Does this have an estimate?
- [x] Does this have acceptance criteria?
- [x] Does this have sufficient context (wireframes, business rules, etc.)?
- [x] Is this traceable to a feature, epic, or PI?

## User Story

As a CAMPD API user I want to see a list of transactions types so that I can include the correct data in my query.

## Acceptance Criteria
Given that a user has made a GET request to the  transactions type endpoint
When the request is executed
Then the system displays all  transactions types in JSON format and for each record displays:

- Transaction type code  
- Transaction type description 

### Does this Story require 508 Testing?
- [ ] Implementing a new authoring tool that will produce code / content for the application
- [ ] Implementing a 3rd party widget that will be part of the application
- [ ] Adding new / updating custom functionality 
- [ ] Adding new / updating a page template
- [ ] Adding new / replacing existing functionality (unless the replacement has already passed 508 testing)
- [ ] Full Site Review prior to a scheduled EPA 508 review
- [ ] Planning a video that will appear on the application
- [ ] Producing a video that will appear on the application
- [ ] Producing Training materials
- [ ] Producing documents

## Context

- want to display a list of all the transaction types 
(User journeys, design mockups, architectural diagrams, business rules, data element definitions)

From: Mark Hayward <Mark.Hayward@erg.com> 
Sent: Tuesday, April 13, 2021 9:35 AM
To: Lageman, Travis <Lageman.Travis@epa.gov>; Ann Albrecht <Ann.Albrecht@erg.com>
Cc: Desantis, Laurel <Desantis.Laurel@epa.gov>; Kuhns, Jason <Kuhns.Jason@epa.gov>; Stetsyuk, Karina <Stetsyuk.Karina@epa.gov>; Gill, Sandeep <Gill.Sandeep@epa.gov>
Subject: RE: Lookup code table for Allowance Transactions

Travis,

CAMDNATS.TTRANTYPE lists the possible transaction types for transactions recorded in the CAMDNATS.TTRANSACT table.  CAMDNATS only includes legacy OTC and NBP data.

CAMDAMS.TRANSACTION_TYPE_CODE lists the possible transaction types for transactions recorded in the CAMDAMS.TRANSACTION table.  This covers all transactions for all other programs (ARP, CAIR, CSAPR, any new program that might come along).

ALLOCATION_IND in CAMDAMS.TRANSACTION_TYPE_CODE has 3 possible values.  
•	0 – transaction type that is not an allocation
•	1 – transaction type that is an allocation
•	-1 – transaction type that acts as a reversal or surrender of an allocation
o	These are exclusively ARP transaction types that were used during ARP Phase 1
	DI            Reallocation Surrender
	RP           Phase 2 Early Reduction Payback
	RS           Reduced Utilization Payback
	SP           State Cap Payback
	SS           Substitution Payback
	ST           Substitution Termination
	TM         Reduced Utilization Termination

The ALLOCATION_IND field is used during the AMPD data mart loading on CAMDPUB when populating the “Compliance Year Allowances Allocated” for compliance query results.

It is also used during the running of compliance in CBS and CSA when determining which allowances to deduct using FIFO.

Mark

From: Lageman, Travis <Lageman.Travis@epa.gov> 
Sent: Monday, April 12, 2021 2:03 PM
To: Mark Hayward <Mark.Hayward@erg.com>; Ann Albrecht <Ann.Albrecht@erg.com>
Cc: Desantis, Laurel <Desantis.Laurel@epa.gov>; Kuhns, Jason <Kuhns.Jason@epa.gov>; Stetsyuk, Karina <Stetsyuk.Karina@epa.gov>; Gill, Sandeep <Gill.Sandeep@epa.gov>
Subject: Lookup code table for Allowance Transactions

CAUTION: Don't open links or attachments unless you recognize the sender and know they are safe.
Hi Mark & Ann,

Hope you had a nice weekend. I have another database question. I was looking for the transaction type code table and realized there are two different tables, one under the TRANSACTION_TYPE_CODE in CAMDAMS (which I believe is the current) and then one under TTRANTYPE in CAMDNATS (the older). I believe these are both necessary to get the full list of transaction types in the database. Is this correct or is there additional context I should be aware of?

I also noticed that there are some -1 values in the ALLOCATION_IND field for TRANSACTION_TYPE_CODE which I am confused as to what that identifies as opposed to 0 or 1, which make sense.

Thanks in advance,

Travis Lageman
Environmental Policy Analyst
Clean Air Markets Division
U.S. Environmental Protection Agency
P: 202-564-0520

## Testing
Test Case:[ TC321](https://cvpcorp.testquality.com/project/11428/plan/21550/test/298146)
Tester: @JanellC 

### Definition of Done
- [x] Have all Acceptance Criteria been agreed to and validated by the scrum team?
- [x] Has the code been unit tested and peer reviewed?
- [x] Have the functional tests been executed?
- [x] Have all defects been reviewed and dispositioned, resolved, or deferred?
- [ ] Has the user story been accepted by the Product Owner?
- [x] Is the user story ready to be deployed to the test/staging environments?

**Comment by JanellC** [comment]
TC321 passed @tlageman-epa 
https://easey-dev.app.cloud.gov/api/master-data-mgmt/transaction-types
